test/test_port.cpp

```diff
@@ -138,6 +143,11 @@ TEST_F (PortUnitTest, PortCheckIteratorRepeatCyclic)
    EXPECT_EQ (pf_port_get_next_repeat_cyclic (&port_iterator), 2);
    EXPECT_EQ (pf_port_get_next_repeat_cyclic (&port_iterator), 3);
 
+   /* Verify behaviour if the interator for unknown reason has
+    * value that is out of port range */
+   port_iterator.next_port = 10;
+   EXPECT_EQ (pf_port_get_next_repeat_cyclic (&port_iterator), 1);
```

Comment:
Lägg till
EXPECT_EQ (pf_port_get_next (&port_iterator), 1);

Vi vill testa `.next_port` också, inte bara returvärdet.

Reply:
Jag fattar inte. Menar du:
   /* Verify behaviour if the interator for unknown reason has
    * value that is out of port range */
   port_iterator.next_port = 10;
   EXPECT_EQ (pf_port_get_next_repeat_cyclic (&port_iterator), 1);
   EXPECT_EQ (pf_port_get_next_repeat_cyclic (&port_iterator), 2); <------ NEW ASSERT

Reply:
Jag ändrade implementationen på pf_port_get_next_repeat_cyclic().

Reply:
Snyggt!